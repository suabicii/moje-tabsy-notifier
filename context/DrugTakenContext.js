import React, {createContext, useContext} from "react";

const DrugTakenContext = createContext([]);

export const useDrugTakenContext = () => useContext(DrugTakenContext);

export default DrugTakenContext;