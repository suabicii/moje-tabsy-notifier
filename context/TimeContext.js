import React, {createContext} from "react";

export const TimeContext = createContext(undefined);