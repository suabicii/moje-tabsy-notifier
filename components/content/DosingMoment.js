import React, {useContext} from "react";
import {Button, Paragraph} from "react-native-paper";
import {View} from "react-native";
import {DrugTakenContext} from "../../context/DrugTakenContext";

function DosingMoment({drugName, hour, id}) {
    const {drugTakenChecker, setDrugTakenChecker} = useContext(DrugTakenContext);
    const handleConfirmDose = drugName => {

    };

    // const [hour, minutes] = value.split(':');
    // const dosingDateTime = dayjs().hour(hour).minute(minutes);
    // console.log(dayjs().isSameOrAfter(dosingDateTime));

    return (
        <View
            style={{
                alignItems: "center",
                flexDirection: "row",
                justifyContent: "center",
                marginBottom: 10
            }}
        >
            <Paragraph style={{textAlign: "center", fontSize: 16}}>
                – {`${hour}`}
            </Paragraph>
            <Button
                testID={id}
                style={{
                    backgroundColor: "#6cc3d5",
                    color: "#e8e8e8",
                    marginLeft: 5
                }}
                contentStyle={{flexDirection: "row-reverse"}}
                // disabled={true}
                icon="check"
                mode="contained"
                compact={true}
                onPress={() => {
                    handleConfirmDose(drugName);
                    // setDosingMomentsToShow(
                    //     dosingMomentsToShow.filter(dosingMoment => dosingMoment[0] !== key)
                    // );
                    setDrugTakenChecker([...drugTakenChecker, id]);
                    // Todo: zapis do AsyncStorage
                }}
            >
                Już zażyłem/-am
            </Button>
        </View>
    );
}

export default DosingMoment;