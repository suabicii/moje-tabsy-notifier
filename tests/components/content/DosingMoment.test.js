import React from "react";
import renderer from "react-test-renderer";
import {DrugTakenContext} from "../../../context/DrugTakenContext";
import DosingMoment from "../../../components/content/DosingMoment";
import {drugList} from "../../fixtures/drugList";

const drugTakenChecker = [];
const setDrugTakenChecker = jest.fn();
const mockUseContext = jest.fn().mockImplementation(() => ({drugTakenChecker, setDrugTakenChecker}));

React.useContext = mockUseContext;

function WrappedComponent() {
    const drug = drugList[0];

    return (
        <DrugTakenContext.Provider value={{drugTakenChecker, setDrugTakenChecker}}>
            <DosingMoment
                drugName={drug.name}
                hour={`${Object.values(drug.dosingMoments)[0]}`}
                id={`${drug.name}_${Object.keys(drug.dosingMoments)[0]}`}
            />
        </DrugTakenContext.Provider>
    );
}

it('should correctly render single DosingMoment component', () => {
    const tree = renderer.create(<WrappedComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
});