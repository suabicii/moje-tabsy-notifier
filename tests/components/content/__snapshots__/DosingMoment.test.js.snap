// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render single DosingMoment component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 14,
              "letterSpacing": 0.25,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            Object {
              "fontSize": 16,
              "textAlign": "center",
            },
          ],
        ],
      ]
    }
  >
    – 
    07:00
  </Text>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#6cc3d5",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "color": "#e8e8e8",
        "elevation": 2,
        "marginLeft": 5,
        "minWidth": "auto",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
      testID="Xanax_hour1"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "flexDirection": "row-reverse",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": -4,
              "marginRight": 12,
            }
          }
        >
          <Text />
        </View>
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                Object {
                  "marginHorizontal": 8,
                },
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Już zażyłem/-am
        </Text>
      </View>
    </View>
  </View>
</View>
`;
